perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1433 @@
+
+class HybridTopologyFactory(object):
+    """
+    This class generates a hybrid topology based on a perses topology proposal. This class treats atoms
+    in the resulting hybrid system as being from one of four classes:
+
+    unique_old_atom : these atoms are not mapped and only present in the old system. Their interactions will be on for
+        lambda=0, off for lambda=1
+    unique_new_atom : these atoms are not mapped and only present in the new system. Their interactions will be off
+        for lambda=0, on for lambda=1
+    core_atom : these atoms are mapped, and are part of a residue that is changing. Their interactions will be those
+        corresponding to the old system at lambda=0, and those corresponding to the new system at lambda=1
+    environment_atom : these atoms are mapped, and are not part of a changing residue. Their interactions are always
+        on and are alchemically unmodified.
+
+    Properties
+    ----------
+    hybrid_system : openmm.System
+        The hybrid system for simulation
+    new_to_hybrid_atom_map : dict of int : int
+        The mapping of new system atoms to hybrid atoms
+    old_to_hybrid_atom_map : dict of int : int
+        The mapping of old system atoms to hybrid atoms
+    hybrid_positions : [n, 3] np.ndarray
+        The positions of the hybrid system
+    hybrid_topology : mdtraj.Topology
+        The topology of the hybrid system
+    omm_hybrid_topology : openmm.app.Topology
+        The OpenMM topology object corresponding to the hybrid system
+    """
+
+    _known_forces = {'HarmonicBondForce', 'HarmonicAngleForce', 'PeriodicTorsionForce', 'NonbondedForce', 'MonteCarloBarostat'}
+
+    def __init__(self, topology_proposal, current_positions, new_positions, use_dispersion_correction=False):
+        """
+        Initialize the Hybrid topology factory.
+
+        Parameters
+        ----------
+        topology_proposal : perses.rjmc.topology_proposal.TopologyProposal object
+            TopologyProposal object rendered by the ProposalEngine
+        current_positions : [n,3] np.ndarray of float
+            The positions of the "old system"
+        new_positions : [m,3] np.ndarray of float
+            The positions of the "new system"
+        use_dispersion_correction : bool, default False
+            Whether to use the long range correction in the custom sterics force. This is very expensive for NCMC.
+        """
+        self._topology_proposal = topology_proposal
+        self._old_system = copy.deepcopy(topology_proposal.old_system)
+        self._new_system = copy.deepcopy(topology_proposal.new_system)
+        self._old_to_hybrid_map = {}
+        self._new_to_hybrid_map = {}
+        self._hybrid_system_forces = dict()
+        self._old_positions = current_positions
+        self._new_positions = new_positions
+
+        self._use_dispersion_correction = use_dispersion_correction
+
+        self.softcore_alpha=0.5
+        self.softcore_beta=12*unit.angstrom**2
```

Comment:
By unitless, I mean that softcore_beta is not just lacking units, but uses the LJ effective `sigma` to set the scale of the interaction. See here:
https://github.com/choderalab/openmmtools/blob/master/openmmtools/alchemy.py#L1472